tests/Aql/ShortestPathExecutorTest.cpp

```diff
@@ -175,762 +185,265 @@ struct TestShortestPathOptions : public ShortestPathOptions {
-                                  std::move(constTarget), false);
-}
+  void TestExecutor(ShortestPathExecutorInfos& infos, AqlItemBlockInputRange& input) {
+    // This fetcher will not be called!
+    // After Execute is done this fetcher shall be removed, the Executor does not need it anymore!
+    // This will fetch everything now, unless we give a small enough atMost
 
-TEST_F(ShortestPathExecutorTest,
-       NotWaiting_WithMultipleRows_UsingVertexOutputOnly_UsingConstantSourceInput_UsingRegisterTargetInput) {
-  RunTestWithMultipleRowsUpstream(false, std::move(constSource), std::move(regTarget), false);
-}
+    // We use a copy here because skipRowsRange modifies the the call
+    auto skipCall = AqlCall{parameters._call};
 
-TEST_F(ShortestPathExecutorTest,
-       NotWaiting_WithMultipleRows_UsingVertexOutputOnly_UsingRegisterSourceInput_UsingConstantTargetInput) {
-  RunTestWithMultipleRowsUpstream(false, std::move(regSource), std::move(constTarget), false);
-}
+    auto const [skipState, skipped, resultSkipCall] = testee.skipRowsRange(input, skipCall);
 
-TEST_F(ShortestPathExecutorTest,
-       NotWaiting_WithMultipleRows_UsingVertexOutputOnly_UsingRegisterSourceInput_UsingRegisterTargetInput) {
-  RunTestWithMultipleRowsUpstream(false, std::move(regSource), std::move(regTarget), false);
-}
+    // TODO: Do we want to assert more here?
+    EXPECT_TRUE(skipState == ExecutorState::HASMORE || skipState == ExecutorState::DONE);
 
-TEST_F(ShortestPathExecutorTest,
-       NotWaiting_WithMultipleRows_UsingVertexAndEdgeOutput_UsingConstantSourceInput_UsingConstantTargetInput) {
-  RunTestWithMultipleRowsUpstream(false, std::move(constSource),
-                                  std::move(constTarget), true);
-}
+    // TODO: What (is supposed to) happen(s) if skip already returned DONE?
+    auto const [produceState, stats, resultProduceCall] =
+        testee.produceRows(input, output);
 
-TEST_F(ShortestPathExecutorTest,
-       NotWaiting_WithMultipleRows_UsingVertexAndEdgeOutput_UsingConstantSourceInput_UsingRegisterTargetInput) {
-  RunTestWithMultipleRowsUpstream(false, std::move(constSource), std::move(regTarget), true);
-}
+    // TODO: should we make a way to assert the state?
+    EXPECT_TRUE(produceState == ExecutorState::HASMORE || produceState == ExecutorState::DONE);
 
-TEST_F(ShortestPathExecutorTest,
-       NotWaiting_WithMultipleRows_UsingVertexAndEdgeOutput_UsingRegisterSourceInput_UsingConstantTargetInput) {
-  RunTestWithMultipleRowsUpstream(false, std::move(regSource), std::move(constTarget), true);
-}
+    ValidateResult(output, skipped);
+  }
+};
 
-TEST_F(ShortestPathExecutorTest,
-       NotWaiting_WithMultipleRows_UsingVertexAndEdgeOutput_UsingRegisterSourceInput_UsingRegisterTargetInput) {
-  RunTestWithMultipleRowsUpstream(false, std::move(regSource), std::move(regTarget), true);
-}
+TEST_P(ShortestPathExecutorTest, the_test) { TestExecutor(infos, input); }
```

Comment:
Can you please add a comment how this test works, what does it test and why is it enough?
If you read it long enough it gets clear how it works, but that is rather obfuscated.

Also please add (a/some) test(s) for fullCount.
With fullCount you will get into skip/get/fullCount situation, where I think a bug is hidden.

Also can you add tests for multiple counts?

produce => outputFull, halfPathExists
produce(again) => halfPath continued.